public/app/plugins/panel/timeseries/plugins/OutsideRangePlugin.tsx

```diff
@@ -1,34 +1,43 @@
-import React, { useLayoutEffect, useRef } from 'react';
-import uPlot from 'uplot';
+import React, { useLayoutEffect, useRef, useState } from 'react';
+import uPlot, { TypedArray, Scale } from 'uplot';
 
-import { TimeRange, AbsoluteTimeRange } from '@grafana/data';
+import { AbsoluteTimeRange } from '@grafana/data';
 import { UPlotConfigBuilder, Button } from '@grafana/ui';
 
 interface ThresholdControlsPluginProps {
   config: UPlotConfigBuilder;
-  range: TimeRange;
   onChangeTimeRange: (timeRange: AbsoluteTimeRange) => void;
 }
 
-export const OutsideRangePlugin: React.FC<ThresholdControlsPluginProps> = ({ config, range, onChangeTimeRange }) => {
+export const OutsideRangePlugin: React.FC<ThresholdControlsPluginProps> = ({ config, onChangeTimeRange }) => {
   const plotInstance = useRef<uPlot>();
+  const [timevalues, setTimeValues] = useState<number[] | TypedArray>([]);
+  const [timeRange, setTimeRange] = useState<Scale | undefined>();
 
   useLayoutEffect(() => {
     config.addHook('init', (u) => {
       plotInstance.current = u;
     });
+
+    config.addHook('draw', (u) => {
```

Comment:
not sure if this is a bad idea, but `setScale` and `setData` didn't seem to work on this branch

Reply:
yes, draw hook fires on every resized pixel, so not good.

Reply:
i think i got it in [be94c48](https://github.com/grafana/grafana/pull/49633/commits/be94c486508da27e0f73b80a7d7b175df41e0b9f) using `setScale` by itself